packages/desktop-client/src/components/accounts/TransactionsTable.js

```diff
@@ -1787,6 +1837,120 @@ export let TransactionTable = forwardRef((props, ref) => {
     id => splitsExpanded.dispatch({ type: 'toggle-split', id }),
     [splitsExpanded.dispatch],
   );
+  let [field, setField] = useState(null);
+  let [ascDesc, setAscDesc] = useState('desc');
+
+  function onSortTable(e) {
+    let transform =
+      e.currentTarget.innerText.toLowerCase() === 'payment' ||
+      e.currentTarget.innerText.toLowerCase() === 'deposit'
+        ? 'amount'
+        : e.currentTarget.innerText.toLowerCase();
+    if (transform === field) {
+      ascDesc === 'desc' ? setAscDesc('asc') : setAscDesc('desc');
+    } else {
+      setField(transform);
+      transform === 'amount' || transform === 'date'
+        ? setAscDesc('asc')
+        : setAscDesc('desc');
+    }
+  }
+
+  let category = props.categoryGroups.flatMap(cat => {
+    return cat.categories;
+  });
+  let fields = [field] + 's';
```

Comment:
Why are you implicitly converting an array to a string here?

Reply:
Do you mean `let fields = [field] + 's';` ?
Sorry, `field ` was never an array. I mistakenly had the brakets in there. I've fixed it.